pkg/util/utils_linux.go

```diff
@@ -110,18 +107,7 @@ func AddPrivilegedDevices(g *generate.Generator, systemdMode bool) error {
 				Source:      d.Path,
 				Options:     []string{"slave", "nosuid", "noexec", "rw", "rbind"},
 			}
-
-			/* The following devices should not be mounted in rootless containers:
-			 *
-			 *   /dev/ptmx: The host-provided /dev/ptmx should not be shared to
-			 *              the rootless containers for security reasons, and
-			 *              the container runtime will create it for us
-			 *              anyway (ln -s /dev/pts/ptmx /dev/ptmx);
-			 *   /dev/tty[0-9]+: Prevent the container from taking over the host's
-			 *                   virtual consoles, even when not in systemd mode
-			 *                   for backwards compatibility.
-			 */
-			if d.Path == "/dev/ptmx" || isVirtualConsoleDevice(d.Path) {
+			if d.Path == "/dev/ptmx" || strings.HasPrefix(d.Path, "/dev/tty") {
```

Comment:
The cause could be that `isVirtualConsoleDevice` evaluates to false for `/dev/tty` before `strings.HasPrefix(d.Path, "/dev/tty")` evaluated to true for it.

Reply:
This might also cause an issue in non-privileged mode that has the same change (it's not reverted in this PR).

Reply:
@fho would you mind opening a PR for upstream?

Perhaps I can just cherry-pick the fix if that proves to be obvious

Reply:
I'm hesitant to create a PR, to skip mounting /dev/tty because I don't know how to reason it.
Why must /dev/tty not be mounted and why mounting it would cause an error like: `merged/dev/tty: no such device or address`?
/dev/tty is not a pseudo terminal. We actually even had a short discussion about it when that change was introduced: https://github.com/containers/podman/pull/17055#discussion_r1065810750.
https://systemd.io/CONTAINER_INTERFACE/, https://github.com/containers/podman/issues/15878 and https://github.com/dcermak/podman/commit/5a2405ae1b3a51a7fb1f01de89bd6b2c60416f08 does not mention that /dev/tty must not be mounted. Am I missing something?


I also tried to reproduce  the bug locally with podman e7878726f68bb14c67f0905e9215e581ab89012e but could not:
`bin/podman run --rm -d --privileged quay.io/libpod/testimage:20221018 top` works fine. The container starts and keeps running.
I'm using crun 1.7.2 locally, not the runc version from the bug report though.
Is that maybe an incompatibility with runc or a bug in that runc version?